doc/design/empty_tensor_slice.md

```diff
@@ -0,0 +1,32 @@
+### Empty Tensor Design
```

Comment:
I think we don't need to create a new .md file here; instead, we can move the content here to https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/framework/tensor.md